datafusion/src/datasource/mod.rs

```diff
@@ -193,9 +192,13 @@ pub fn get_statistics_with_limit(
 
     let mut num_rows = 0;
     let mut num_files = 0;
+    let mut is_exact = true;
     for file in &all_files {
         num_files += 1;
         let file_stats = &file.statistics;
+        if !file_stats.is_exact {
```

Comment:
```suggestion
        is_exact &= file_stats.is_exact;
```

Reply:
nice optim! bit harder to read for people like me who don't have a compiler in their head, but I like it 😉